var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -144,6 +144,9 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     variant("kokkos", default=False, description="Activates support for kokkos and kokkos-kernels")
     variant("fortran", default=True, description="Activates fortran support")
 
+    # https://github.com/spack/spack/issues/37416
+    conflicts("^rocprim@5.3.0", when="@3.19.1 +rocm")
```

Comment:
```suggestion
    conflicts("^rocprim@5.3.0:5.3.2", when="+rocm")
```

I believe the fix is in rocm@5.3.3.

 And its unlikely any version of PETSc would work with these versions of rocm..

cc: @jczhang07

Reply:
Just committed these changes! Thank you.